### PR TITLE
Prevent reload to same page in Pagination

### LINKED ISSUE #3161: Pagination buttons in mobile view malfunction
https://care.coronasafe.in/facility
The "previous" button in mobile view reloads the page even if we are on the 1st page.

https://www.loom.com/share/2ad83863942f472ea42b44defa6ccfc4

### PR DESCRIPTION
**Pull request by Pranshu1902**
Fixes #3161 
http://localhost:4000/facility

[Screencast from 18-07-22 03:26:44 PM IST.webm](https://user-images.githubusercontent.com/70687348/179488202-0f068386-d113-4990-a749-b9d3b477aec2.webm)

### CODE REVIEW

File: src/Components/Common/Pagination.tsx [hunk 1]
```diff
@@ -89,8 +89,10 @@ const Pagination = (props: PaginationProps) => {
         break;
     }
 
-    setCurrentPage(newPage);
-    onChange(newPage, rowsPerPage);
+    if (newPage >= 1 && newPage <= totalPage) {
+      setCurrentPage(newPage);
+      onChange(newPage, rowsPerPage);
+    }
```
Comment (developedBySJ) [inline src/Components/Common/Pagination.tsx]: LGTM

**Review by developedBySJ — CHANGES_REQUESTED** [review]
@Pranshu1902 
- Use the `button` tag instead of `div`
- Add `disable` prop to button (eg: disable "prev" button if a user is on the first page)

https://github.com/coronasafe/care_fe/blob/ff03007dbd5c6565afe829789a04d8221cde5dfc/src/Components/Common/Pagination.tsx#L131-L136

**Comment by Pranshu1902** [comment]
@developedBySJ done

**Comment by nihal467** [comment]
test approved

File: src/Components/Common/Pagination.tsx [hunk 2]
```diff
@@ -128,18 +126,20 @@ const Pagination = (props: PaginationProps) => {
   return (
     <div className="mx-auto mb-4">
       <div className="flex-1 flex justify-between sm:hidden">
-        <div
+        <button
+          {...(currentPage === 1 ? { disabled: true } : {})}
```
Comment (developedBySJ) [inline src/Components/Common/Pagination.tsx]:
```suggestion
        disabled={currentPage === 1}
```

File: src/Components/Common/Pagination.tsx [hunk 3]
```diff
@@ -128,18 +126,20 @@ const Pagination = (props: PaginationProps) => {
...
           onClick={() => handleChangePage("<")}
           className="relative inline-flex items-center px-4 py-2 border border-gray-300 text-sm leading-5 font-medium rounded-md text-gray-700 bg-white hover:text-gray-500 focus:outline-none focus:ring-primary focus:border-primary-300 active:bg-gray-100 active:text-gray-700 transition ease-in-out duration-150"
         >
           Previous
-        </div>
-        <div
+        </button>
+        <button
+          {...(currentPage === totalPage ? { disabled: true } : {})}
```
Comment (developedBySJ) [inline src/Components/Common/Pagination.tsx]:
```suggestion
          disabled={currentPage === totalPage}
```

**Review by developedBySJ — CHANGES_REQUESTED** [review]

**Comment by Pranshu1902** [comment]
Fixed in #3204

**Pull request closed without merging**